eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway/DoctrineDatabase.php

```diff
@@ -344,8 +344,20 @@ public function moveSubtreeNodes(array $sourceNodeData, array $destinationNodeDa
                 ->set(
                     $this->handler->quoteColumn('parent_node_id'),
                     $query->bindValue($newParentId)
-                )
-                ->where(
+                );
+
+            // Mark tree as invisible if destinatio is, and/or is hidden.
+            if ($destinationNodeData['is_hidden'] || $destinationNodeData['is_invisible']) {
+                $query->set(
+                    $this->handler->quoteColumn('is_hidden'),
+                    $query->bindValue(0)
```

Comment:
Is it correct to unset hidden? I would expect any explicitly hidden location to stay hidden until explicitly unhidden.